File: our-moments/src/main/java/com/gravity/ourmoments/entity/Comment.java
package com.gravity.ourmoments.entity;

import lombok.Data;
import java.time.LocalDateTime;

@Data
public class Comment {
    private Long commentId;
    private Long postId;
    private Long userId;
    private String content;
    private Integer position; // 评论在文章中的位置
    private LocalDateTime createTime;
    private LocalDateTime updateTime;

    // For joining data
    private User author;
}

File: our-moments-android/app/src/main/java/com/gravity/ourmoments/data/model/Comment.kt
package com.gravity.ourmoments.data.model

import kotlinx.serialization.Serializable

@Serializable
data class Comment(
    val commentId: Long,
    val content: String,
    val post: Post,
    val author: User,
    val createTime: String,
    val updateTime: String? = null
)